src/cskvp.rs

```diff
@@ -177,65 +298,64 @@ impl<'a> Parser<'a> {
             .unwrap_or(self.rest.len());
         let rest = self.rest.split_off(idx);
         let mut val = mem::replace(&mut self.rest, rest);
-        val.trim_inplace();
-        val
+        let self_range = self.range.clone();
+        self.range.start += idx;
+
+        let len = val.len();
+        val.trim_start_inplace();
+        let trimmed_start = len - val.len();
+        val.trim_end_inplace();
+        let trimmed_end = len - trimmed_start - val.len();
+        let range = Range {
+            start: self_range.start + trimmed_start,
+            end: self_range.start + idx - trimmed_end,
+        };
+        (val, range)
     }
 
-    fn next_single(&mut self, delimiters: &[char]) -> Cow<'a, str> {
+    fn next_single(
+        &mut self, delimiters: &[char], diagnostics: &mut Diagnostics<'a>,
+    ) -> Result<(Cow<'a, str>, Range<usize>), Diagnostic> {
+        let len = self.rest.len();
         self.rest.trim_start_inplace();
-        if self.rest.starts_with('"') { self.next_quoted() } else { self.next_unquoted(delimiters) }
+        self.range.start += len - self.rest.len();
+        if self.rest.starts_with('"') {
+            self.next_quoted(diagnostics)
+        } else {
+            Ok(self.next_unquoted(delimiters))
+        }
     }
 
     fn skip_delimiter(&mut self) -> Option<char> {
+        let len = self.rest.len();
         self.rest.trim_start_inplace();
+        self.range.start += len - self.rest.len();
         let delim = self.rest.chars().next();
         if delim.is_some() {
             self.rest.truncate_start(1);
+            self.range.start += 1;
         }
         delim
     }
 }
 
-impl<'a> Iterator for Parser<'a> {
-    type Item = Value<'a>;
-
-    fn next(&mut self) -> Option<Value<'a>> {
-        if self.rest.is_empty() {
-            return None;
-        }
-
-        let mut key = self.next_single(&['=', ',']);
-        key.trim_inplace();
-
-        let delim = self.skip_delimiter();
-
-        if let Some('=') = delim {
-            let res = Some(Value::Double(key, self.next_single(&[','])));
-
-            let delim = self.skip_delimiter();
-            assert!(delim == Some(',') || delim == None, "invalid comma seperated key value pair");
-            res
-        } else {
-            // TODO: error handling
-            assert!(delim == Some(',') || delim == None, "invalid comma seperated value");
-            Some(Value::Single(key))
-        }
-    }
-}
-
 #[cfg(test)]
 mod test {
```

Comment:
The test suite doesn't compile right now. Problably has been updated before some interface changes and then not run again?

Reply:
Should I still do this or are you fixing this in #77 ?

Reply:
I think it should be fixed here.